### ISSUE TITLE
[Phase 3] Implement Isolation Forest Anomaly Detection

### ISSUE THREAD

**Issue by Neon-Nomad** [Neon-Nomad/Amunet-AI#18]
## Overview
Build lightweight ML model for anomaly detection.

## Requirements
- [ ] Train Isolation Forest model on baseline data
- [ ] Detect traffic anomalies (spikes, unusual patterns)
- [ ] Real-time scoring (< 100ms per event)
- [ ] Anomaly score: 0.0 (normal) to 1.0 (anomalous)
- [ ] Alert on score > 0.8
- [ ] False positive tracking and model tuning

## Implementation (Python)
```python
from sklearn.ensemble import IsolationForest
import numpy as np

# Train model on baseline
baseline_data = load_baseline_data()  # 2 weeks of metrics
model = IsolationForest(contamination=0.1, random_state=42)
model.fit(baseline_data)

# Real-time scoring
def score_anomaly(metrics: dict) -> float:
    features = extract_features(metrics)
    score = model.decision_function([features])[0]
    normalized = 1.0 / (1.0 + np.exp(score))  # Sigmoid
    return normalized

# Example
metrics = {"requests_per_minute": 5000, "error_rate": 0.5}
anomaly_score = score_anomaly(metrics)  # → 0.92 (HIGH)
```

## Success Criteria
- Model trained on 2 weeks of data
- Detects traffic spikes with 80%+ precision
- < 5% false positive rate
- Real-time scoring integrated into control plane

**Priority**: P2
**Effort**: 1 week
**Labels**: ml, phase-3, anomaly-detection
